### PR TITLE
reorganize and add build timeout script

### PR DESCRIPTION
**Pull request by clytwynec**
Some details about what changed:
* `duplicate_ghprb_jobs/` --> `jenkins/`
* Interactions with the jenkins api are moved out into `jenkins/job.py` from `deduper.py`.
* `deduper.py` now just deals with determining which builds are duplicates and using `job.py` to stop them.
* `timeout.py` was added to deal with determining which builds have been running too long and using `job.py` to stop them.
* Tests are updated/added to account for the above changes.

Usage:
* `deduper.py` usage remains the same.
* To use `timeout.py`: 
```python jenkins/timeout.py -u [username] -t [api_token] -j [job_url] --timeout [timeout_in_minutes]```

### CODE REVIEW

**Comment by clytwynec** [comment]
@benpatterson @jzoldak Please review

File: jenkins/job.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    def __init__(self, job_url, username, token):
+        self.job_url = job_url
+        self.auth = (username, token)
+
+    def get_json(self):
+        """
+        Get build data for a given job_url.
+
+        :Returns:
+            A python dict from the jeknins api response including:
```
Comment (jzoldak) [inline jenkins/job.py]: typo jenkins

File: jenkins/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            ('http://my_base_url.com', 'the_extra_part'),
+            ('http://my_base_url.com', '/the_extra_part'),
+            ('http://my_base_url.com/', 'the_extra_part'),
+            ('http://my_base_url.com/', '/the_extra_part'),
+        ]
+
+        for i in inputs:
+            returned = append_url(*i)
+            self.assertEqual(expected, returned,
+                             msg="{e} != {r}\nInputs: {i}".format(
```
Comment (jzoldak) [inline jenkins/tests/test_helpers.py]: This indenting will be more readable if you follow the [edx-platform python line-breaking guidelines](https://github.com/edx/edx-platform/wiki/Python-Guidelines#breaking-long-lines).

File: jenkins/tests/test_job.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from mock import patch
+from requests.exceptions import HTTPError
+from unittest import TestCase
+
+from jenkins.tests.helpers import mock_response, sample_data
+from jenkins.job import JenkinsJob
+
+
+class JenkinsJobTestCase(TestCase):
+
```
Comment (jzoldak) [inline jenkins/tests/test_job.py]: Too many empty lines?

File: jenkins/timeout.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    def get_stuck_builds(self, data):
+        """
+        Return build data for currently running builds.
+
+        :Args:
+            data: the return value of self.get_json()
+
+        :Returns:
+            build_data: ???
```
Comment (jzoldak) [inline jenkins/timeout.py]: Did you mean to circle back and add detail here?

**Comment by jzoldak** [comment]
Couple style points, but overall the code looks good. :+1:

File: jenkins/timeout.py (hunk 4 above)
Comment (clytwynec) [inline jenkins/timeout.py]: yes I did. I'll do that now

**Pull request merged**